feathr_project/registry/registry/database.py

```diff
@@ -0,0 +1,84 @@
+from abc import ABC, abstractmethod
+import threading
+from distutils.log import debug, warn
+import os
+import pymssql
+
+
+providers = []
+
+class DbConnection(ABC):
+    @abstractmethod
+    def execute(self, sql: str, *args, **kwargs) -> list[dict]:
+        pass
+
+def quote(id):
+    if isinstance(id, str):
+        return f"'{id}'"
+    else:
+        return ",".join([f"'{i}'" for i in id])
+
+
+def parse_conn_str(s: str) -> dict:
+    """
+    TODO: Not a sound and safe implementation
```

Comment:
Put TODO in comments? In the method description, we should say the connection string format that we are expecting (guess it should be standard connection string and maybe we can put some reference)

Reply:
It's just ADO connection string, I'm just not sure if the implementation covered all cases, it works if you simply copy the connection string from the Azure portal, but not thoroughly tested for all other cases.
This value is not from the external input so there is no security issue.